src/foam/nanos/auth/UserAndGroupAuthService.js

```diff
@@ -258,9 +248,33 @@ foam.CLASS({
     {
       name: 'validatePassword',
       javaCode: `
-        if ( SafetyUtil.isEmpty(potentialPassword) || ! (Pattern.compile(PASSWORD_VALIDATE_REGEX)).matcher(potentialPassword).matches() ) {
-          throw new RuntimeException(PASSWORD_VALIDATION_ERROR_MESSAGE);
+        PasswordPolicy passwordPolicy = null;
+
+        // Retrieve the password policy from the user and group when available
+        if ( user != null ) {
+          Group ancestor = user.findGroup(x);
+          if ( ancestor != null ) {
+            throw new AuthenticationException("Group not found");
```

Comment:
I think this should be a `RuntimeException` instead. `AuthenticationException` means the user needs to authenticate (sign in) to do what they're trying to do. In this case the data is corrupted, so signing in won't change anything.

We should also add a logger statement here before throwing and record the user's id and their group.